src/main/java/frc/robot/systems/DriveFSMSystem.java

```diff
@@ -115,14 +115,43 @@ public FSMState getCurrentState() {
 		return currentState;
 	}
 	/**
-	 * Reset this system to its start state. This may be called from mode init
+	 * Reset this system to its autonomous state. This may be called from mode init
 	 * when the robot is enabled.
 	 *
 	 * Note this is distinct from the one-time initialization in the constructor
 	 * as it may be called multiple times in a boot cycle,
 	 * Ex. if the robot is enabled, disabled, then reenabled.
 	 */
-	public void reset() {
+	public void resetAutonomous() {
```

Comment:
I'd much prefer there by a generic "reset" method for tasks that have to be performed on any reset, and the other stuff should be put in the transition to Teleop state so that we're not duplicating the reset code. Also, setting currentState shouldn't happen anywhere other than via nextState().

Reply:
How else would we set the starting state of the robot if we don't do it in the reset method?

Reply:
I meant between autonomous and teleop.